WordPressKit/BlockEditorSettingsServiceRemote.swift

```diff
@@ -43,30 +28,18 @@ public extension BlockEditorSettingsServiceRemote {
 public extension BlockEditorSettingsServiceRemote {
     typealias BlockEditorSettingsCompletionHandler = (Swift.Result<RemoteBlockEditorSettings?, Error>) -> Void
 
-    func fetchBlockEditorSettings(forSiteID siteID: Int?, _ completion: @escaping BlockEditorSettingsCompletionHandler) {
-        let requestPath = "/wp-block-editor/v1/settings"
-        let parameters: [String: AnyObject] = ["context": "mobile" as AnyObject]
-        let modifiedPath = remoteAPI.requestPath(fromOrgPath: requestPath, with: siteID)
-
-        remoteAPI.GET(modifiedPath, parameters: parameters) { [weak self] (result, _) in
-            guard let `self` = self else { return }
-            switch result {
-            case .success(let response):
-                self.processBlockEditorSettingsResponse(response) { blockEditorSettings in
-                    completion(.success(blockEditorSettings))
+    func fetchBlockEditorSettings(completion: @escaping BlockEditorSettingsCompletionHandler) {
+        Task { @MainActor in
+            let result = await self.remoteAPI.get(path: "/wp-block-editor/v1/settings", parameters: ["context": "mobile"], type: RemoteBlockEditorSettings.self)
+                .map { settings -> RemoteBlockEditorSettings? in settings }
```

Comment:
This might be a change for later on, but it seems like the only reason the `RemoteBlockEditorSettings` the API call returns is wrapped in an `Optional` here is because that's the type the "legacy" `BlockEditorSettingsCompletionHandler` expects?

If so, and since we're making breaking changes, I wonder whether we could make the type in the completion typealias non-optional. It would remove this step, and I suspect also simplify caller code by removing the need to deal with `Result` _and_ `Optional`

Reply:
It's not just for `fetchBlockEditorSettings`'s syntax level backward compatibility, but most importantly runtime behaviour backward compatibility.

`/wp-block-editor/v1/settings` endpoint returns a JSON that contains many properties. The `fetchBlockEditorSettings` function only cares about a subset of these properties. See the model type [`RemoteBlockEditorSettings`](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/trunk/WordPressKit/RemoteBlockEditorSettings.swift), which contains mostly non-optional properties.

If the JSON response contains all the properties declared in the Swift type, this function of course should return a model instance.

However, if this API returns 200 with valid "block editor settings" JSON response, which _does not_ contains all the non-optional proprieties declared in the `RemoteBlockEditorSettings` type, the function should return a nil, instead of a "decoding error".

That's the behaviour of the existing function. And the refactored implementation (mainly the line here and the code below that maps decoding error to a nil successful result) maintains the same runtime behaviour for backward compatibility.